src/main/java/seedu/address/model/person/patient/Weight.java

```diff
@@ -32,7 +32,8 @@ public Weight(String weight) {
      * Returns true if a given string is a valid weight.
      */
     public static boolean isValidWeight(String test) {
-        return test.matches(VALIDATION_REGEX);
+        test = test.trim();
+        return test.matches(VALIDATION_REGEX) && !test.equals("0") && !test.equals("0.0");
```

Comment:
We may want to use regex to encode the checks for 0

Reply:
You may want to try `^(?!0(\.0+)?$)\d+(\.\d{1,})?$`. I am not sure if it works but may be worth a shot.

Reply:
This regex does not fit the requirement to have at most 1 decimal place, but I will try to find a solution for this

Reply:
Added as issue as non-urgent